junifer/markers/crossatlas.py

```diff
@@ -0,0 +1,135 @@
+"""Provide marker class to calculate cross-atlas FC."""
+
+# Authors: Leonard Sasse <l.sasse@fz-juelich.de>
+#          Kaustubh R. Patil <k.patil@fz-juelich.de>
+# License: AGPL
+
+from typing import Dict, List, Optional
+
+import pandas as pd
+
+from ..api.decorators import register_marker
+from ..utils import logger
+from .base import BaseMarker
+from .parcel_aggregation import ParcelAggregation
+from .utils import _correlate_dataframes
+
+
+@register_marker
+class CrossAtlasFC(BaseMarker):
+    """Class for calculating parcel-wise correlations between two atlases.
+
+    Parameters
+    ----------
+    atlas_one : str
+        The name of the first atlas.
+    atlas_two : str
+        The name of the second atlas.
+    aggregation_method : str, optional
+        The aggregation method (default "mean").
+    correlation_method : str or callable
+        any method that can be passed to:
+        https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.corr.html
+
+    """
+
+    def __init__(
+        self,
+        atlas_one: str,
+        atlas_two: str,
+        aggregation_method: str = "mean",
+        correlation_method: str = "pearson",
+        name: str = None,
+    ) -> None:
+        """Initialize the class."""
+        self.atlas_one = atlas_one
+        self.atlas_two = atlas_two
+        self.aggregation_method = aggregation_method
+        self.correlation_method = correlation_method
+        super().__init__(on=["BOLD"], name=name)
+
+    def get_valid_inputs(self) -> List[str]:
+        """Get valid data types for input.
+
+        Returns
+        -------
+        list of str
+            The list of data types that can be used as input for this marker
+
+        """
+        return ["BOLD"]
+
+    def get_output_kind(self, input: List[str]) -> List[str]:
+        """Get output kind.
+
+        Parameters
+        ----------
+        input : list of str
+            The input to the marker. The list must contain the
+            available Junifer Data dictionary keys.
+
+        Returns
+        -------
+        list of str
+            The updated list of output kinds, as storage possibilities.
+
+        """
+        return ["matrix"]
+
+    def store(self, kind: str, out, storage) -> None:
+        """Store.
+
+        Parameters
+        ----------
+        kind
```

Comment:
This should be "matrix" I think, since the output is not BOLD but a matrix, but I get your point.

Reply:
This would be "BOLD" as data type is "BOLD". This makes more sense for markers operating on many data types, for example, `ParcelAggregation`. So, based on the data type, you call the `storage.store()` with the correct storage kind i.e., "timeseries" or "matrix" or "table".

Reply:
i think this is not quite intuitive to me --> the input data type cant really be used to determine the output storage kind (i.e. BOLD would be stored actually as a time series). The output data is not BOLD but functional connectivity.

Reply:
The marker implementation decides which format ({"timeseries", "matrix, "table"}) the data type ({"BOLD", ...} ) should be stored in. Your marker only works on "BOLD", so it stays as "BOLD" and then you decide how you store it, for your case it's "matrix" and not "timeseries". So, we are not generalizing it for all markers but making it specific to a marker.

Reply:
ok, makes sense (to some degree), will change it.